### DISCUSSION TITLE
Error serializing in NextJS/Stripe

### DISCUSSION THREAD

**Discussion by 2-GO** [community/community#61872] [Programming Help]
### Select Topic Area

Bug

### Body

Hello Community, 

I encountered the following error "Error: Error serializing `.product` returned from `getStaticProps` in "/products/[id]".
Reason: `undefined` cannot be serialized as JSON. Please use `null` or omit this value." 

This happened when I tried creating individual product pages for my e-commerce website. I have looked at a few solutions suggesting to JSON.parse(JSON.stringify("")) but none have worked. 

Does anyone have any suggestions on how I can go about fixing this error?

My code in "/products/[id]"

```
import Image from "next/image";
import { stripe } from "src/utils/stripe";

export default function ProductPage({ product }) {
    return (
        <div>
             Product
        </div>
        );
}

export async function getStaticPaths() {
    const inventory = await stripe.products.list();
    const paths = inventory.data.map((product) => ({
        params: { id: product.id },
    }));

    return {
        paths, 
        fallback: "blocking"
    };
}

export async function getStaticProps({ params }) {
    const inventory = await stripe.products.list({
        expand: ["data.default_price"],
      });
      const products = inventory.data.map((product) => {
        const price = product.default_price;
        return {
          currency: price.currency,
          id: product.id,
          name: product.name,
          price: price.unit_amount,
          image: product.images[0],
        };
      });
      const product = products.find((product) => product.id === params.id);

      return {
        props: {
            product,
        },
        revalidate: 60 * 60,
    };
} 
```

please let me know if I am missing any relevant information to this particular problem.

**Comment by queenofcorgis** [comment]
Thanks for posting in the GitHub Community, @2-GO ! 

We’ve moved your post to our new Programming Help 🧑‍💻 category, which is more appropriate for this type of discussion.

Please review our [guidelines](https://github.com/orgs/community/discussions/51384) about the Programming Help category for more information.

**Reply by 2-GO** [reply]
Ah, I appreciate that @queenofcorgis . I will look into the guidelines for future posts.
